### PR TITLE
Update markdown parser to goldmark to fix visual issues

### PR DESCRIPTION
**Pull request by SuperAuguste**
## Background

I was informed that Sourcegraph's server-side markdown infrastructure was using an unmaintained library, causing the following visual issue:

image

## Test plan

Tested locally with https://github.com/mxstbr/markdown-test-file to ensure general compliance.

And of course, here's the original malady remedied:
image
image

Hope this is to your satisfaction!

### CODE REVIEW

File: cmd/frontend/internal/markdown/markdown.go [hunk 1]
```diff
@@ -26,6 +30,17 @@ func Render(content string) string {
...
+		goldmark.WithParserOptions(
+			parser.WithAutoHeadingID(),
+		),
+		goldmark.WithRendererOptions(
+			html.WithUnsafe(),
+		),
+	)
+
+	var buf bytes.Buffer
+	md.Convert([]byte(content), &buf)
```
Comment (SuperAuguste) [inline cmd/frontend/internal/markdown/markdown.go:44]: I'm slightly concerned by the error discard here; how should a critical error be handled in this scenario? An empty string return? Or should it be left as-is?

**Comment by jdorfman** [comment]
Needs color :)

File: cmd/frontend/internal/markdown/markdown.go [hunk 2]
```diff
@@ -23,9 +29,28 @@ func Render(content string) string {
 		policy.AllowAttrs("aria-hidden").Matching(regexp.MustCompile(`^true$`)).OnElements("a")
 		policy.AllowAttrs("type").Matching(regexp.MustCompile(`^checkbox$`)).OnElements("input")
 		policy.AllowAttrs("checked", "disabled").Matching(regexp.MustCompile(`^$`)).OnElements("input")
-		policy.AllowAttrs("class").Matching(regexp.MustCompile("^language-[a-zA-Z0-9]+$")).OnElements("code")
+		policy.AllowAttrs("class").OnElements("pre", "code", "span")
```
Comment (SuperAuguste) [inline cmd/frontend/internal/markdown/markdown.go:32]: This should still be safe, feel free to restrict this further if necessary.

**Review by beyang — COMMENTED** [review]
@SuperAuguste I pulled your changes and added a few changes on top of it to address the comments below. Here's the new PR: https://github.com/sourcegraph/sourcegraph/pull/46960.

Will merge that pending a few internal approvals. We can go ahead and close this PR for now, thanks for the changes!

**Pull request closed without merging**